src/Stratis.Bitcoin.Features.Miner/PowBlockDefinition.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Features.Consensus;
+using Stratis.Bitcoin.Features.Consensus.Interfaces;
+using Stratis.Bitcoin.Features.MemoryPool;
+using Stratis.Bitcoin.Features.MemoryPool.Interfaces;
+using Stratis.Bitcoin.Mining;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Miner
+{
+    public class PowBlockDefinition : BlockDefinition
+    {
+        /// <summary>Instance logger.</summary>
+        private readonly ILogger logger;
+
+        // Container for tracking updates to ancestor feerate as we include (parent)
+        // transactions in a block.
+        public class TxMemPoolModifiedEntry
+        {
+            public TxMemPoolModifiedEntry(TxMempoolEntry entry)
+            {
+                this.iter = entry;
+                this.SizeWithAncestors = entry.SizeWithAncestors;
+                this.ModFeesWithAncestors = entry.ModFeesWithAncestors;
+                this.SigOpCostWithAncestors = entry.SigOpCostWithAncestors;
+            }
+
+            public TxMempoolEntry iter;
+
+            public long SizeWithAncestors;
+
+            public Money ModFeesWithAncestors;
+
+            public long SigOpCostWithAncestors;
+        }
+
+        // This matches the calculation in CompareTxMemPoolEntryByAncestorFee,
+        // except operating on CTxMemPoolModifiedEntry.
+        // TODO: Refactor to avoid duplication of this logic.
+        public class CompareModifiedEntry : IComparer<TxMemPoolModifiedEntry>
+        {
+            public int Compare(TxMemPoolModifiedEntry a, TxMemPoolModifiedEntry b)
+            {
+                Money f1 = a.ModFeesWithAncestors * b.SizeWithAncestors;
+                Money f2 = b.ModFeesWithAncestors * a.SizeWithAncestors;
+
+                if (f1 == f2)
+                    return TxMempool.CompareIteratorByHash.InnerCompare(a.iter, b.iter);
+
+                return f1 > f2 ? 1 : -1;
```

Comment:
Would returning bool work, or is it best to return 1/-1?

Reply:
Will revisit in subsequent PRs.